x/auth/tx/decoder.go

```diff
@@ -9,7 +9,8 @@ import (
 )
 
 // DefaultTxDecoder returns a default protobuf TxDecoder using the provided Marshaler.
-func DefaultTxDecoder(cdc *codec.ProtoCodec) sdk.TxDecoder {
+func DefaultTxDecoder(cdc *codec.ProtoCodec, unpacker ServiceRequestUnpacker) sdk.TxDecoder {
+	// TODO unpack ServiceMsg requests
```

Comment:
At some point in this function we do `err = cdc.UnmarshalBinaryBare(raw.BodyBytes, &body)`.

If the body bytes contain a ServiceMsg-encoded msg, then this will throw an error. I'm not sure how the unpacker can help in that case

Reply:
Unless you mean changing now `TxBody.Messages []ServiceMsg` in the proto files?